dbms/src/AggregateFunctions/AggregateFunctionTopK.h

```diff
@@ -119,7 +127,14 @@ class AggregateFunctionTopK
 /// Generic implementation, it uses serialized representation as object descriptor.
 struct AggregateFunctionTopKGenericData
 {
-    using Set = SpaceSaving<StringRef, StringRefHash>;
+    using Set = SpaceSaving
+    <
+        std::string,
+        StringRef,
+        StringRefHash,
+        HashTableGrower<TOP_K_DEFAULT_DEGREE>,
+        HashTableAllocatorWithStackMemory<sizeof(StringRef) * (1 << (TOP_K_DEFAULT_DEGREE/2))>
```

Comment:
Yes, it makes a lot of guessing. By default there are 10 elements in the sorted vector and the hashtable, but it's variable on function call time. So the allocator reserves up to 32 elements on stack (less with saved hash), which I estimate is enough for most use cases (aligned to 8-bytes). 16 would be another okay value.